src/webgpu/web_platform/util.ts

```diff
@@ -689,3 +705,44 @@ export async function GetSourceFromImageFile(
     }
   }
 }
+
+/**
+ * Create HTMLImageElement and load image file and waits for it to be loaded.
+ * Returns a promise which resolves after `callback` (which may be async) completes.
+ *
+ * @param imageName An valid imageName in kkImageInfo table .
+ * @param callback Function to call when HTMLImageElement is loaded.
+ *
+ */
+export function loadImageFileAndRun(
+  test: GPUTest,
+  imageName: ImageName,
+  callback: (image: HTMLImageElement) => unknown | Promise<unknown>
```

Comment:
~Is the mixed callback + promise thing necessary, instead of just returning `Promise<HTMLImageElement>`? I know we have it for videos but I think that was only because video playback can advance in the background.~ Oh, wait, I suppose GIFs and SVGs can do that too. Never mind, this is probably safer.